### PR TITLE
Simplify invariant output

### PR DESCRIPTION
**Pull request by elazarg**
Group variables in equivalence classes, and print details only about one  of them.

## Release Notes

- **New Features**
  - Enhanced functionality for extracting and representing constraints in the `SplitDBM` class.
  - Introduced a new comparison method for variable ordering in the `variable_t` class.

- **Bug Fixes**
  - Adjusted test cases to ensure correct handling of arithmetic operations, particularly regarding value assignments between `svalue` and `uvalue`.

- **Documentation**
  - Improved clarity in test case descriptions and error messages related to division by zero and uninitialized registers.

- **Tests**
  - Comprehensive updates to test cases across various operations, ensuring robust coverage of edge cases and correct logic in assignments.

### CODE REVIEW

**Comment by elazarg** [comment]
@Alan-Jowett & @dthaler what do you think about the results - [this one](https://github.com/vbpf/ebpf-verifier/pull/798/files#diff-f5ab54cde6303074ef0cad6ab2b0fc2fc390afb429dac036addc34533bc9e29dL227-L268) for example? Should this be behind a flag?

File: test-data/movsx.yaml [hunk 1]
```diff
@@ -146,11 +137,9 @@ code:
 post:
   - r1.type=number
   - r1.svalue=[32768, 32770]
-  - r1.uvalue=[32768, 32770]
   - r1.svalue=r1.uvalue
```
Comment (dthaler) [inline test-data/movsx.yaml:140]: If you have "pc[1]=r0.svalue=r0.uvalue" elsewhere then why wouldn't this be something like
"r1.uvalue=r1.svalue=[32768, 32770]"?

**Comment by dthaler** [comment]
> I don't think this needs to be behind a flag. I think it should be fine as the default behavior.

I'm ok either way.  If I'm looking for a particular variable, it's easier to find when in alphabetical order instead of having to look for it somewhere other than at the beginning of a line.  However, it's not a big deal either way.

File: test-data/movsx.yaml (hunk 1 above)
Comment (elazarg) [reply test-data/movsx.yaml:140]: I though it was too much.

**Comment by elazarg** [comment]
> > I don't think this needs to be behind a flag. I think it should be fine as the default behavior.
> 
> I'm ok either way. If I'm looking for a particular variable, it's easier to find when in alphabetical order instead of having to look for it somewhere other than at the beginning of a line. However, it's not a big deal either way.

Maybe leave the equalities in the beginning of the line?
```
r1.svalue=[0, 3]
r2.svalue=r1.svalue  # use only r1.svalue everywhere else
r5.svalue <= r1.svalue + 2
```

File: test-data/add.yaml [hunk 2]
```diff
@@ -130,20 +125,12 @@ post:
...
-  - r7.svalue-r2.svalue<=-7
-  - r7.svalue-r2.uvalue<=-7
-  - r2.svalue-r7.svalue<=11
-  - r2.uvalue-r7.svalue<=11
   - r2.packet_offset-r2.svalue<=-7
-  - r2.packet_offset-r2.uvalue<=-7
   - r2.svalue-r2.packet_offset<=11
-  - r2.uvalue-r2.packet_offset<=11
-  - r2.packet_offset=r7.svalue
+  - r7.svalue=r2.packet_offset
```
Comment (dthaler) [inline test-data/add.yaml:133]: Now i'm wondering why `r7.uvalue=r2.packet_offset` and `r7.uvalue=r7.svalue` weren't invariants before.

File: test-data/full64.yaml [hunk 3]
```diff
@@ -197,22 +194,16 @@ post:
...
 
 code:
   <start>: |
-     assume r1 == r2
+    assume r1 == r2
 
 post:
+  - r1.uvalue=r1.svalue
+  - r2.svalue=r1.svalue
+  - r2.uvalue=r1.svalue
```
Comment (dthaler) [inline test-data/full64.yaml:206]: Should we list invariants in alphabetical order?  Right now the r1's aren't together in order (line 204 and 207 are separated by r2...=...)

File: test-data/add.yaml (hunk 2 above)
Comment (elazarg) [reply test-data/add.yaml:133]: I suspect there was a bug in to_set

File: test-data/full64.yaml (hunk 3 above)
Comment (elazarg) [reply test-data/full64.yaml:206]: The test comparison is order-insensitive

Comment (elazarg) [reply test-data/full64.yaml:206]: It's a good idea to make the tests more consistent though
